backend/controllers/event.js

```diff
@@ -0,0 +1,149 @@
+  const event = await Event.findById(req.params.eventId).exec();
+  if (!event)
+    return res.status(400).json({ error: "Event not found in the DB!" });
+  return res.json(event);
+});
+
+// @desc    getEvents
+// @route   GET /api/events
+// @access  Public
+const getEvents = expressAsyncHandler(async (req, res) => {
+  const today = new Date();
+  const events = await Event.find({ startDate: { $gte: today } });
+  if (events) {
+    res.json(events);
+  } else {
+    res.status(404);
+    throw new Error("Events not found");
+  }
+});
+
+// @desc    addEvent
+// @route   POST /api/events/addEvent
+// @access  Admin
+const addEvent = expressAsyncHandler(async (req, res) => {
+  const { name, desc, startDate, photo, url } = req.body;
+  const event = await Event.create({ name, desc, startDate, photo, url });
+  if (event) {
+    res.json({
+      _id: event._id,
+      name: event.name,
+      desc: event.desc,
+      start: event.startDate,
+      photo: event.startDate,
+      url: event.url,
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid data");
+  }
+});
+
+// @desc    updateEvent
+// @route   PUT /api/events/updateEvent/:eventId
+// @access  Admin
+
+const updateEvent = expressAsyncHandler(async (req, res) => {
+  try {
+    const event = await Event.findById(req.params.eventId);
+    if (event) {
+      await Event.findByIdAndUpdate(
+        { _id: event._id },
+        { $set: req.body },
+        { new: true, UseFindAndModify: false, runValidators: true },
+        (err, updatedEvent) => {
+          if (err) {
+            return res.status(400).json({
+              error: `${err}`,
+            });
+          } else {
+            res.json(updatedEvent);
+          }
+        }
+      );
+    } else {
+      return res.status(404).json({
+        error: "Event with this id doesn't exist",
+      });
+    }
+  } catch (err) {
+    return res.json(err.message);
+  }
+});
+
+// @desc    deletePastEvents
+// @route   DELETE /api/deleteEvent
+// @access  Admin
+const deletePastEvents = expressAsyncHandler(async (req, res) => {
+  const today = new Date();
+  await Event.deleteMany({ startDate: { $lte: today } })
+    .then(() => {
+      res.send("Outdated Events deleted !");
+    })
+    .catch((err) => {
+      console.error(`Operation failed due to ${err}`);
+    });
+});
+
+// @desc    deleteEvent
+// @route   DELETE /api/events/deleteEvent/:eventId
+// @access  Admin
+const deleteEvent = expressAsyncHandler(async (req, res) => {
+  const event = await Event.findById(req.params.eventId);
+  if (event) {
+    await event.remove();
+    res.json({ message: "Event was successfully deleted!" });
+  } else {
+    res.status(404).json({
+      error: "Failed to delete this Event",
+    });
+  }
+});
+
+// @desc    getScrappedEvents
+// @route   GET /api/events/getScrappedEvents
+// @access  Public
+const getScrappedEvents = expressAsyncHandler(async (req, res) => {
```

Comment:
I feel its not future compatible. I mean if we want to add more urls, we need to write the code again for each url, instead we could have used a utility function for this and loop through all urls. Its fine for now. I merging this code.